File: support_tools/swigwin-3.0.12/Examples/test-suite/go/director_detect_runme.go
package main

import "./director_detect"

type MyBar struct {
	val int
} // From director_detect.Bar

func NewMyBar() director_detect.Bar {
	return director_detect.NewDirectorBar(&MyBar{2})
}

func (p *MyBar) Get_value() int {
	p.val++
	return p.val
}

func (p *MyBar) Get_class() director_detect.A {
	p.val++
	return director_detect.NewA()
}

func (p *MyBar) Just_do_it() {
	p.val++
}

func (p *MyBar) Clone() director_detect.Bar {
	return director_detect.NewDirectorBar(&MyBar{p.val})
}

func main() {
	b := NewMyBar()

	f := b.Baseclass()

	v := f.Get_value()
	_ = f.Get_class()
	f.Just_do_it()

	c := b.DirectorInterface().(*MyBar).Clone()
	vc := c.Get_value()

	if (v != 3) || (b.DirectorInterface().(*MyBar).val != 5) || (vc != 6) {
		panic("Bad virtual detection")
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/director_detect_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'director_detect'

class MyBar < Director_detect::Bar
  def initialize(v)
    @val = v
  end

  def get_value
    @val = @val + 1
  end
  
  def get_class
    @val = @val + 1
    Director_detect::A
  end

  def just_do_it
    @val = @val + 1
  end

  def clone
    MyBar.new(@val)
  end

  def val
    @val
  end
end


b = MyBar.new(2)

f = b

v = f.get_value
a = f.get_class
f.just_do_it

c = b.clone
vc = c.get_value

raise RuntimeError if (v != 3) || (b.val != 5) || (vc != 6)

